firmware/steering/kia_soul_ps/steering_control_module.ino

```diff
@@ -334,34 +343,105 @@ void disable_control()
     current_ctrl_state.control_enabled = false;
 
     // Disable the signal interrupt relays
-    digitalWrite(SPOOF_ENGAGE, LOW);
+    digitalWrite( SPOOF_ENGAGE, LOW );
 
-    DEBUG_PRINT("Control disabled");
+    DEBUG_PRINT( "Control disabled" );
 }
 
 
 // *****************************************************
 // Function:    calculate_torque_spoof
-// 
+//
 // Purpose:     Container for hand-tuned empirically determined values
-// 
+//
 //              Values calculated with min/max calibration curve and hand
 //              tuned for neutral balance.
 //              DAC requires 12-bit values = (4096steps/5V = 819.2 steps/V)
-// 
+//
 // Returns:     void
-// 
+//
 // Parameters:  [in] torque - floating point value with the current torque value
 //              [out] torque_spoof - structure containing the integer torque values
-// 
+//
 // *****************************************************
-void calculate_torque_spoof( float torque, struct torque_spoof_t* spoof ) 
+void calculate_torque_spoof( float torque, struct torque_spoof_t* spoof )
 {
     spoof->low = 819.2 * ( 0.0008 * torque + 2.26 );
     spoof->high = 819.2 * ( -0.0008 * torque + 2.5 );
 }
 
 
+// *****************************************************
+// Function:    check_spoof_voltages
+//
+// Purpose:     Check for discrepancies between DAC output and ADC input for
+//              spoof voltages.
+//
+// Returns:     void
+//
+// Parameters:  [out] torque_spoof - struct containing the integer torque values
+//
+// *****************************************************
+void check_spoof_voltages( struct torque_spoof_t* spoof ) // L -> A, H -> B
+{
+
+    uint16_t spoof_a_adc = analogRead( SPOOF_SIGNAL_A );
+    uint16_t spoof_b_adc = analogRead( SPOOF_SIGNAL_B );
+
+    float spoof_a_adc_volts = spoof_a_adc * ( 5.0 / 1023.0 ) + 0.010;
+    float spoof_b_adc_volts = spoof_b_adc * ( 5.0 / 1023.0 ) + 0.010;
+
+    // DAC values passed in from calculate_torque_spoof( )
+    float spoof_a_dac_current_volts = spoof->high * ( 5.0 / 4095.0 );
+    float spoof_b_dac_current_volts = spoof->low * ( 5.0 / 4095.0 );
+
+    // fail criteria. ~ ( ± 96mV )
+    if ( abs( spoof_a_adc_volts - spoof_a_dac_current_volts ) >
+            VOLTAGE_THRESHOLD )
+    {
+        if ( current_ctrl_state.override_flag.voltage_spike_a == 0 )
+        {
+            current_ctrl_state.override_flag.voltage_spike_a = 1;
+        }
+        else
+        {
+            DEBUG_PRINT( "* * ERROR!!  Voltage Discrepancy on Signal A. * *" );
+
+            disable_control( );
+            current_ctrl_state.override_flag.voltage = 1;
+        }
+    }
+    else
+    {
+        current_ctrl_state.override_flag.voltage = 0;
+        current_ctrl_state.override_flag.voltage_spike_a = 0;
+    }
+
+    // fail criteria. ~ ( ± 96mV )
+    if ( abs( spoof_b_adc_volts - spoof_b_dac_current_volts ) >
+            VOLTAGE_THRESHOLD )
+    {
+        if ( current_ctrl_state.override_flag.voltage_spike_b == 0 )
```

Comment:
Can we make this flag more descriptive? Like "voltage_spike_first_occurrence" or something?